### ISSUE TITLE
Add GearVR support!

### ISSUE THREAD

**Issue by JoeMattie** [jbienz/FPVR#1]
Pretty pretty please?!

I can assist, though I have limited experience with Unity.

**Comment by jbienz** [comment]
Hey Joe. Just saw your comment. I'd love to support GearVR, but that means I have to buy one! LOL. I don't mind spending $200 bucks on a device, but it's the additional $600 that has to be spent on the phone.

FYI, for this to work you'd at least need the V2 GearVR. From what I can tell it appears the V2 does have an exposed Micro USB port. It's not clear yet if that Micro USB port is a hub or connects to the phone in passthrough mode. If not, GearVR won't work because the app needs a USB port in order to use the DJI SDK.

**Comment by JoeMattie** [comment]
Why would it need an exposed USB port to work with the SDK?  I've compiled
several test apps using the DJI SDK and they work just fine.  At the
moment, the only problem I'm having is with passing the frames from the
DJIReceivedVideoCallback into Gear's native mode so they can be mapped to a
texture....

**Comment by jbienz** [comment]
Oh, you must be flying a Phantom 1 or 2 then. That's right, I forget that those work over Wifi only. The Phantom 3 and the Inspire 1 (which I have) require a USB port. Instead of connecting to the aircraft over wifi, they connect to the aircraft via the USB port which is connected directly to the controller. They depend on the radio in the controller itself which gives them GREATLY increased range and reliability.

Have you seen this thread about decoding the video that comes in the callback?
http://forum.dji.com/thread-4705-1-1.html

Are you writing your GearVR app in Unity? It might be good to collaborate on this portion of the project. Have you seen the Unity subfolder of this GitHub? You might want to check that out. Though right now I'm building it on Unity for PC (since that's what Oculus needs) you should be able to convert it to Unity for Android. And I could use some help with that. I don't have a video surface in there yet because I've got problems on the PC side. My capture card isn't working with Unity's WebcamTexture. I may have to write my own (which is not going to be fun at all).

**Comment by jbienz** [comment]
P.S. if you manage to get the video decoding working in Unity for Android I'd like to see about shuttling those bits across wifi to the PC version for the Oculus Rift instead of trying to use a capture card. That would be much more convenient, assuming the latency of sending the bits over wifi isn't significantly higher than using the capture card.

**Comment by JoeMattie** [comment]
Yeah I'm running a Phantom 2 Vision+.  I have very little experience with
unity, but I will give it another look.   I have seen that thread, yes.  It
looks pretty straightforward, although it would be nice if mdepasquale
posted an example of a working function
